src/System.Runtime.Extensions/tests/System/IO/PathTests_Unix.cs

```diff
@@ -28,22 +29,32 @@ public void GetFileName_Volume(string path, string expected)
             Assert.Equal(expected, new string(Path.GetFileName(path.AsSpan())));
         }
 
-        [Theory,
-            InlineData("/tmp/", "/tmp"),
-            InlineData("/tmp/", "/tmp/"),
-            InlineData("/", "/"),
-            InlineData("/var/tmp/", "/var/tmp"),
-            InlineData("/var/tmp/", "/var/tmp/"),
-            InlineData("~/", "~"),
-            InlineData("~/", "~/"),
-            InlineData(".tmp/", ".tmp"),
-            InlineData("./tmp/", "./tmp"),
-            InlineData("/home/someuser/sometempdir/", "/home/someuser/sometempdir/"),
-            InlineData("/home/someuser/some tempdir/", "/home/someuser/some tempdir/"),
-            InlineData("/tmp/", null)]
-        public void GetTempPath_SetEnvVar_Unix(string expected, string newTempPath)
+        public static IEnumerable<string[]> GetTempPath_SetEnvVar_Data()
+        {
+            yield return new string[] { "/tmp/", "/tmp" };
+            yield return new string[] { "/tmp/", "/tmp/" };
+            yield return new string[] { "/", "/" };
+            yield return new string[] { "/var/tmp/", "/var/tmp" };
+            yield return new string[] { "/var/tmp/", "/var/tmp/" };
+            yield return new string[] { "~/", "~" };
+            yield return new string[] { "~/", "~/" };
+            yield return new string[] { ".tmp/", ".tmp" };
+            yield return new string[] { "./tmp/", "./tmp" };
+            yield return new string[] { "/home/someuser/sometempdir/", "/home/someuser/sometempdir/" };
+            yield return new string[] { "/home/someuser/some tempdir/", "/home/someuser/some tempdir/" };
+            yield return new string[] { "/tmp/", null };
+        }
+
+        [Fact]
+        public void GetTempPath_SetEnvVar_Unix()
         {
-            GetTempPath_SetEnvVar("TMPDIR", expected, newTempPath);
+            RemoteInvoke(() =>
+            {
+                foreach (string[] tempPath in GetTempPath_SetEnvVar_Data())
```

Comment:
What's the reason for switching from a theory to a fact with a loop?

Reply:
I was not able to pass the arguments from the test ```(string expected, string newTempPath)``` to the remoteInvoke due to Field Marshalling

Reply:
You pass them in as arguments to RemoteInvoke, and it then passes them into the lambda, e.g.
https://github.com/dotnet/corefx/blob/eb892471f8691dad65ff47f5b12ddc8d378ae460/src/System.Net.Http/tests/FunctionalTests/PostScenarioUWPTest.cs#L57